demonstrations/qnspsa.py

```diff
@@ -371,7 +371,7 @@ def get_overlap_tape(qnode, params1, params2):
         for op in op_forward:
             qml.apply(op)
         for op in reversed(op_inv):
-            op.adjoint()
+            qml.adjoint(op)
```

Comment:
```suggestion
            qml.adjoint(copy(op))
```

If this works, it should be fine, but you may need to do this instead.

`qml.adjoint(op)` marks the original operation as "owned", so it won't end up in the circuit.